stats/stats/src/data_source/kinds/local_db/parameters/update/batching/mod.rs

```diff
@@ -203,12 +239,196 @@ mod tests {
                 )],
             ),
         ] {
-            let expected: Vec<_> = expected
-                .into_iter()
-                .map(|r| RangeInclusive::new(r.0, r.1))
-                .collect();
+            let expected: Vec<_> = expected.into_iter().map(|r| r.0..r.1).collect();
             let actual = generate_date_time_ranges(from, to, chrono::Duration::days(30));
             assert_eq!(expected, actual);
         }
+
+        // 1 day batch
+        assert_eq!(
+            vec![day_start(d("2015-07-20"))..day_start(d("2015-07-21"))],
+            generate_date_time_ranges(
+                day_start(d("2015-07-20")),
+                day_start(d("2015-07-21")),
+                chrono::Duration::days(1)
+            )
+        );
+        assert_eq!(
+            vec![
+                day_start(d("2015-07-20"))..day_start(d("2015-07-21")),
+                day_start(d("2015-07-21"))..day_start(d("2015-07-22"))
+            ],
+            generate_date_time_ranges(
+                day_start(d("2015-07-20")),
+                day_start(d("2015-07-22")),
+                chrono::Duration::days(1)
+            )
+        )
+    }
```

Comment:
you only test 1day batch.
I suggest also test 30 days batch and check if month change works correctly

Reply:
but tests for 30 days is right above these?...

Reply:
right